File: backend/src/main/java/com/twitty/backend/service/NewsService.java
package com.twitty.backend.service;

import com.twitty.backend.dto.NewsResponseDTO;
import com.twitty.backend.dto.PostResponseDTO;
import com.twitty.backend.entity.News;

import java.util.List;

public interface NewsService {
    void saveNews(News news);
    List<NewsResponseDTO> getNews(int limit, int offset);
}

File: frontend/src/Services/news.service.js
import WebService from "./webService";

const service = () => {   
  const getNews = async (limit, offset) => {
    let { response, error } = await WebService.get("api/user/news/getall?limit="+limit+"&offset="+offset);

    let resp = response.map((res)=> {
      if(res.newsImg){
        res = {...res, "newsImg": process.env.REACT_APP_BACKEND_DOMAIN + res.newsImg}
      }
  
      if(res.userProfileImg){
        res = {...res, "userProfileImg": process.env.REACT_APP_BACKEND_DOMAIN + res.userProfileImg}
      }
      return res;
    })
    return { response:resp, error };
  };

  return { getNews };
};

const newsService = service();
export default newsService;
